web-extension/background.js

```diff
@@ -37,6 +69,10 @@ function _processMessage(message, sender, _) {
         case 'LOGIN_TAB':
             return _processLoginTabMessage(entry, tab);
 
+        // Does not work on first invocation in firefox due to bug https://bugzilla.mozilla.org/show_bug.cgi?id=1466092
```

Comment:
What do you mean by "first invocation"? To me it behaves the same the first time after opening firefox or on consecutive tries.
I also suspect that this is more due to race-conditions in gopassbridge:
- `browser.tabs.sendMessage(currentTab.id, { type: 'MARK_LOGIN_FIELDS' });` seem to be cause of one instance of this error. Surrounding it with `if (tab.status === 'complete' && tab.url !== 'about:blank')` might be a start. Better would be to try to reuse the `_waitForTabLoaded` logic?
- I also suspect that the `let currentTab;` in gopassbridge.js can be problematic as `currentTab` seems to be set asynchronously but we assume in many places that it is always correctly set? I'm not quite sure what happens during transitions like switching tabs or creating new tabs. Can there be a state where `currentTab` should maybe rather be `null` to indicate that there is no valid tab at this moment?
- It looks like `_processLoginTabMessage` is called to soon after opening the new tab. Waiting a bit in `_waitForTabLoaded` with e.g. `setTimeout(() => resolve(tab), 5000);` seems to confirm this, as it works then. `_waitForTabLoaded` should probably wait for the DOM to be loaded in the tab, if this is possible?

Reply:
Interesting, for me this only happends on first load in firefox, not in chrome. Content scripts are loaded after the DOM is loaded. So this might be the problem. Not sure if you can monitor this from the background script of if retrying is just the way to go.

Reply:
Is anywhere documented what `tab.status === 'complete'` actually means? If this does not probably work event-driven that alternative might be to send some kind of "PING" event between background and content script... 😞

Reply:
I found not much about `tab.status` in the MDN or chrome docs except `Whether the tabs have completed loading.`...

Reply:
@martinhoefling And yes, with the additional check it works fine for me in Firefox 👍

Reply:
Well "completed loading" is not ambiguous right ;-)